test/integ/test_multi_statement.py

```diff
@@ -0,0 +1,268 @@
+    assert results[1] == 98765, "the second result was wrong"
+    assert results[2] == 123456, "the third result was wrong"
+
+
+@pytest.mark.skipolddriver
+def test_insert_select_multi(conn_cnx, db_parameters):
+    """Naive use of multi-statement to check multiple SQL functions."""
+    with conn_cnx() as con:
+        with con.cursor() as cur:
+            cur.execute("alter session set MULTI_STATEMENT_COUNT = 0;")
+            table_name = random_string(5, "test_multi_table_").upper()
+            cur.execute(
+                "use schema {db}.{schema};\n"
+                "create table {name} (aa int);\n"
+                "insert into {name}(aa) values(123456),(98765),(65432);\n"
+                "select aa from {name} order by aa;\n"
+                "drop table {name};".format(
+                    db=db_parameters["database"],
+                    schema=db_parameters["schema"]
+                    if "schema" in db_parameters
+                    else "PUBLIC",
+                    name=table_name,
+                )
+            )
+
+            assert cur.fetchall()[0][0] == "Statement executed successfully."
+            assert cur.nextset()
+            assert cur.fetchall()[0][0] == f"Table {table_name} successfully created."
+            assert cur.nextset()
+            assert cur.fetchall()[0][0] == 3
+            assert cur.nextset()
+            results = [rec[0] for rec in cur]
+            _check_results(cur, results)
+            assert cur.nextset()
+            assert cur.fetchall()[0][0] == f"{table_name} successfully dropped."
+
+
+@pytest.mark.skipolddriver
+def test_binding_multi(conn_cnx):
+    """Tests using pyformat and qmark style bindings with multi-statement"""
+    test_string = "select {s}; select {s}, {s}; select {s}, {s}, {s};"
+    for style in ("pyformat", "qmark"):
+        with conn_cnx(paramstyle=style) as con:
+            with con.cursor() as cur:
+                sql = test_string.format(s="%s" if style == "pyformat" else "?")
+                cur.execute(sql, (10, 20, 30, "a", "b", "c"), _num_statements=3)
+                assert cur.fetchall()[0] == (10,)
+                assert cur.nextset()
+                assert cur.fetchall()[0] == (20, 30)
+                assert cur.nextset()
+                assert cur.fetchall()[0] == ("a", "b", "c")
+                assert cur.nextset() is None
+                assert not cur.fetchall()
+
+
+@pytest.mark.skipolddriver
+def test_async_exec_multi(conn_cnx):
+    """Tests whether async execution query works within a multi-statement"""
+    with conn_cnx() as con:
+        with con.cursor() as cur:
+            cur.execute_async(
+                "select 1; select 2; select count(*) from table(generator(timeLimit => 5)); select 'b';",
+                _num_statements=4,
+            )
+            q_id = cur.sfqid
+            status = con.get_query_status(q_id)
+            assert con.is_still_running(status)
+    time.sleep(5)
```

Comment:
Please don't add such a long sleep into automated tests, make it shorter and do a while loop to wait